doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+    };
+
+    /// IMPLEMENTATION NOTE: ResultMatch from AppInstallerRepositorySearch.h
+    /// A single result from the search.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass ResultMatch
+    {
+        /// The package found by the search request.
+        CatalogPackage CatalogPackage { get; };
+
+        /// The highest order field on which the package matched the search.
+        PackageMatchFilter MatchCriteria { get; };
+    }
+    /// IMPLEMENTATION NOTE: SearchResult from AppInstallerRepositorySearch.h
+    /// Search result data returned from FindPackages
+        [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass FindPackagesResult
+    {
+        /// The full set of results from the search.
+        Windows.Foundation.Collections.IVectorView<ResultMatch> Matches { get; };
+
+        /// If true, the results were truncated by the given SearchRequest::MaximumResults.
+        /// USAGE NOTE: Windows Package Manager does not support result pagination, there is no way to continue 
+        /// getting more results.
+        Boolean IsTruncated{ get; };
+    }
+
+    /// Options for FindPackages
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass FindPackagesOptions
+    {
+        FindPackagesOptions();
+
+        /// DESIGN NOTE: 
+        /// This class maps to SearchRequest from  AppInstallerRepositorySearch.h 
+        /// That class is a container for data used to filter the available manifests in an app catalog.
+        /// Its properties can be thought of as:
+        /// (Query || Inclusions...) && Filters...
+        /// If Query and Inclusions are both empty, the starting data set will be the entire database.
+        /// Everything && Filters...
+        /// That has been translated in this api so that 
+        /// Query is PackageMatchField::AppCatalogDefined and PackageMatchFilter.IsAdditive = true
+        /// Inclusions are PackageMatchFilter.IsAdditive = true
+        /// Filters are PackageMatchFilter.IsAdditive = false
+
+        /// Filters to find packages
+        /// USAGE NOTE: Only one filter with PackageMatchField::AppCatalogDefined is allowed.
+        Windows.Foundation.Collections.IVector<PackageMatchFilter> Filters { get; };
+
+        /// Restricts the length of the returned results to the specified count.
+        UInt32 ResultLimit;
+
+        /// Overrides the default search behavior for this search if the catalog is a composite catalog.
+        /// IMPLEMENTATION NOTE: Windows Package Manager currently only supports setting the search behavior when 
+        /// creating the source so this is new work but is expected to be very small.
+        CompositeSearchBehavior CompositeSearchBehavior;
+    }
+
+    /// IMPLEMENTATION NOTE: ISource from AppInstallerRepositorySource.h
+    /// A catalog for searching for packages

```

Comment:
If it's a catalog for searching for *packages*, why is it called an *app* catalog instead of a *package* catalog?

Windows Runtime has three types of app catalogs.

* AppServiceCatalog
* AppExtensionCatalog
* PackageCatalog

Reply:
A good question that I hope we argue about in a couple minutes.

Reply:
RECOMMEND: `PackageCatalog` here and `PackageInstaller` later  (replace this app catalog with package catalog term globally.)